File: java/src/main/java/com/maestro/agent/OpenCodeAgent.java
package com.maestro.agent;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.CompletableFuture;

public class OpenCodeAgent {
    private Map<String, String> customCommands = new HashMap<>();
    private boolean lspEnabled = true;

    public void loadCustomCommand(String id, String template) {
        customCommands.put(id, template);
        System.out.println("Loaded custom command macro: " + id);
    }

    public CompletableFuture<String> executeCustomCommand(String id, Map<String, String> args) {
        return CompletableFuture.supplyAsync(() -> {
            if (!customCommands.containsKey(id)) {
                throw new RuntimeException("Custom command not found: " + id);
            }
            String template = customCommands.get(id);
            for (Map.Entry<String, String> entry : args.entrySet()) {
                template = template.replace("$" + entry.getKey(), entry.getValue());
            }
            try { Thread.sleep(100); } catch (InterruptedException e) {}
            return "Executed macro '" + id + "' resulting in: " + template;
        });
    }

    public CompletableFuture<String> requestLspDiagnostics(String filepath) {
        return CompletableFuture.supplyAsync(() -> {
            if (!lspEnabled) {
                throw new RuntimeException("LSP is currently disabled");
            }
            try { Thread.sleep(150); } catch (InterruptedException e) {}
            return "[{\"file\": \"" + filepath + "\", \"line\": 42, \"msg\": \"simulated lsp error\"}]";
        });
    }
}

File: src/server/agents/OpenCodeAgent.ts
export class OpenCodeAgent {
	private customCommands: Map<string, string> = new Map();
	public lspEnabled: boolean = true;

	constructor() {}

	public loadCustomCommand(id: string, template: string): void {
		this.customCommands.set(id, template);
		console.log(`Loaded custom command macro: ${id}`);
	}

	public async executeCustomCommand(id: string, args: Record<string, string>): Promise<string> {
		return new Promise((resolve, reject) => {
			let template = this.customCommands.get(id);
			if (!template) {
				reject(new Error(`Custom command not found: ${id}`));
				return;
			}

			for (const [key, value] of Object.entries(args)) {
				template = template.split(`$${key}`).join(value);
			}

			setTimeout(() => {
				resolve(`Executed macro '${id}' resulting in: ${template}`);
			}, 100);
		});
	}

	public async requestLspDiagnostics(filepath: string): Promise<string> {
		return new Promise((resolve, reject) => {
			if (!this.lspEnabled) {
				reject(new Error('LSP is currently disabled'));
				return;
			}
			setTimeout(() => {
				resolve(`[{"file": "${filepath}", "line": 42, "msg": "simulated lsp error"}]`);
			}, 150);
		});
	}
}
